dev/benchmarks/macrobenchmarks/README.md

```diff
@@ -4,54 +4,36 @@ Performance benchmarks use either flutter drive or the web benchmark harness.
 
 ## Mobile benchmarks
 
-### Cull opacity benchmark
-
-To run the cull opacity benchmark on a device:
-
-```
-flutter drive --profile test_driver/cull_opacity_perf.dart
-```
-
-Results should be in the file `build/cull_opacity_perf.timeline_summary.json`.
-
-More detailed logs should be in `build/cull_opacity_perf.timeline.json`.
-
-### Cubic bezier benchmark
-
-To run the cubic bezier benchmark on a device:
+To run a mobile benchmark on a device:
 
 ```
-flutter drive --profile test_driver/cubic_bezier_perf.dart
+flutter drive --profile -t run_app.dart --driver test_driver/[test_name]_test.dart
 ```
 
-Results should be in the file `build/cubic_bezier_perf.timeline_summary.json`.
+Results should be in the file `build/[test_name].timeline_summary.json`.
 
-More detailed logs should be in `build/cubic_bezier_perf.timeline.json`.
+More detailed logs should be in `build/[test_name].timeline.json`.
 
-### Backdrop filter benchmark
+The key `[test_name]` can be:
 
-To run the backdrop filter benchmark on a device:
+- `animated_placeholder_perf`
+- `backdrop_filter_perf`
+- `cull_opacity_perf`
+- `fading_child_animation_perf`
+- `imagefiltered_transform_animation_perf`
+- `multi_widget_construction_perf`
+- `picture_cache_perf`
+- `post_backdrop_filter_perf`
+- `simple_animation_perf`
+- `textfield_perf`
 
-```
-flutter drive --profile test_driver/backdrop_filter_perf.dart
-```
-
-Results should be in the file `build/backdrop_filter_perf.timeline_summary.json`.
-
-More detailed logs should be in `build/backdrop_filter_perf.timeline.json`.
-
-### Post Backdrop filter benchmark
-
-To run the post-backdrop filter benchmark on a device:
+`cubic_bezier_perf` is a special case for SkSL shader warm-up, which has its own
```

Comment:
`cubic_bezier_perf` is not just for the SkSL shader wram-up. As you noticed before, `cubic_bezier_perf__timeline_summary` is a normal driver test without SkSL, and only `cubic_bezier_perf_sksl_warmup__timeline_summary` involves SkSL.

Therefore, I think one should still be able to use `flutter drive -t test_driver/run_app.dart --driver cubic_bezier_perf_test.dart` to run the non-SkSL test locally.

I wonder if `cubic_bezier_perf_sksl_warmup__timeline_summary` had any troubles if you did the change above?

Reply:
> I wonder if `cubic_bezier_perf_sksl_warmup__timeline_summary` had any troubles if you did the change above?

It runs as expected, but it make difference in therms of the performance metrics. Locally I don't see significant difference except for `worst_frame_rasterizer_time_millis`, but that one is usually flaky. 

I still added the change but let me know if I should revert it.

Reply:
It's fine to change `createCubicBezierPerfTest` as is now. You may need to also update `createCubicBezierPerfSkSLWarmupTest` and make sure that `cubic_bezier_perf_sksl_warmup__timeline_summary` devicelab tasks run successfully locally.

For this README file, I don't think you have to give a special treatment to `cubic_bezier_perf`. You should be able to just include it as a normal `[test_name]`.

Finally, what `worst_frame_rasterizer_time_millis` difference numbers are you seeing? Is it for `cubic_bezier_perf__timeline_summary` or `cubic_bezier_perf_sksl_warmup__timeline_summary`?

Reply:
> Finally, what `worst_frame_rasterizer_time_millis` difference numbers are you seeing? Is it for `cubic_bezier_perf__timeline_summary` or `cubic_bezier_perf_sksl_warmup__timeline_summary`?

40ms for using run_app and 12 ms for using the warmup version.

Reply:
I don't see anything for `createCubicBezierPerfSkSLWarmupTest` that needs updating.

Reply:
Oh, you didn't remove `cubic_bezier_perf.dart`. I think you can remove it and just let `createCubicBezierPerfSkSLWarmupTest` use `run_app.dart` too, and update `createCubicBezierPerfSkSLWarmupTest`.

In terms of the performance difference, 40ms/12ms might be within the [noise range][traces]. Do you have detailed numbers, like is it within the range of 80ms-120ms for the non-warm-up version?

[traces]: https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3Dworst_frame_rasterizer_time_millis%26test%3Dcubic_bezier_perf__timeline_summary%26test%3Dcubic_bezier_perf_sksl_warmup__timeline_summary

Reply:
I should have mentioned, my numbers are tested on Pixel4 rather than Moto G4. Moto is always flaky on my local side due to #59263. Test takes quite some time. I'll test a couple more times before updates. And the difference I mean both using `PerfTest` (or equivalently `flutter drive`) but with `run_app.dart` or `cubic_bezier_perf.dart` (this includes some codes for warm up).

Reply:
Update: run 4 times for using `run_app.dart` and `cubic_bezier_perf.dart` for `createCubicBezierPerfTest` on MotoG4. No significant difference is observed, but `worst_frame_rasterizer_time_millis` looks slightly better for `cubic_bezier_perf.dart` (89.064, 83.331, 94.123, 83.505 vs 97.895, 87.275, 100.392, 106.045).